store/postgres/src/deployment.rs

```diff
@@ -1010,3 +1010,31 @@ pub fn set_entity_count(
         .execute(conn)?;
     Ok(())
 }
+
+pub fn set_earliest_block(
+    conn: &PgConnection,
+    site: &Site,
+    earliest_block: BlockNumber,
+) -> Result<(), StoreError> {
+    use subgraph_deployment as d;
+
+    update(d::table.filter(d::id.eq(site.id)))
+        .set(d::earliest_block_number.eq(earliest_block))
+        .execute(conn)?;
+    Ok(())
+}
+
+/// Lock the row for `site` in `subgraph_deployment` for update. This lock
+/// is used to coordinate the changes that the subgraph writer makes with
+/// changes that other parts of the system, in particular, pruning make
+//  see also: deployment-lock-for-update
```

Comment:
Since this doesn't return a handle, I assume it locks for the duration of the transaction? Worth adding a note or linking to relevant PG documentation.

Reply:
Locks in PG are always to the end of the txn - there's no way to unlock those internal lock (pg_advisory_lock is a different story, but they can span txns) In any event, I updated the comment